v4-client/src/main/java/com/adobe/platform/core/identity/services/cosmosdb/client/V4AsyncCosmosDbClient.java

```diff
@@ -138,7 +136,11 @@ public SimpleResponse readDocuments(String collectionName, List<String> docIdLis
         options.setMaxBufferedItemCount(1000000);
 
 
-        FeedResponse<JsonNode> results = ItemOperationsBridge.readManyAsync(getCosmosContainerOrLoad(collectionName), docIdList.stream().map(id -> Pair.of(id, new PartitionKey(id))).collect(Collectors.toList()), options, JsonNode.class).block();
+        List<CosmosItemIdentity> itemIdentities =
+            docIdList.stream().map(id ->
+                    new CosmosItemIdentity(new PartitionKey(id), id)).collect(Collectors.toList());
+
+        FeedResponse<JsonNode> results = ItemOperationsBridge.readManyAsync(getCosmosContainerOrLoad(collectionName), itemIdentities, options, JsonNode.class).block();
```

Comment:
`ItemOperationsBridge.readManyAsync` can change to `container.readMany()`

Reply:
@moderakh  one difference between the container.readMany vs ItemOperationsBridge.readManyAsync is that I can not pass cosmosQueryRequestOptions to container.readMany.  

maxDegreeOfParallelism is -1 in container.readMany.

Or should we add cosmosQueryRequestOptions to container.ReadMany?

Reply:
ok, let's stick to what you already have. makes sense, thanks for the explanation. please merge.

Reply:
Thanks~